tuxemon/core/item/item.py

```diff
@@ -276,18 +276,18 @@ class with the name of the effect you define in item.db.
 
 
 def decode_inventory(session, owner, data):
-    """ Reconstruct inventory from save_data
+    """ Reconstruct inventory from a save_data dict
 
     :param session:
     :param owner:
-    :param data: save data
+    :param data: save data inventory
     :type data: Dictionary
 
     :rtype: Dictionary
     :returns: New inventory
     """
     out = {}
-    for slug, quant in (data.get('inventory') or {}).items():
+    for slug, quant in data.items():
```

Comment:
i haven't looked closely, but just a reminder to check callers of `decode_inventory` and make sure they pass in the inventory dictionary

Reply:
pretty sure i did that, but i'll make sure to look again.

Reply:
expected my comment to post when i resolved....
i *had* missed two places, which revealed themselves in my testing.
These have been fixed too